src/main/scala/ai/privado/languageEngine/ruby/tagger/schema/RubyMongoSchemaMapper.scala

```diff
@@ -0,0 +1,209 @@
+package ai.privado.languageEngine.ruby.tagger.schema
+
+import ai.privado.cache.{DatabaseDetailsCache, RuleCache}
+import ai.privado.model.{
+  CatLevelOne,
+  Constants,
+  DatabaseColumn,
+  DatabaseDetails,
+  DatabaseSchema,
+  DatabaseTable,
+  FilterProperty,
+  InternalTag,
+  Language,
+  NodeType,
+  RuleInfo
+}
+import ai.privado.tagger.PrivadoSimpleCpgPass
+import ai.privado.utility.ConfigParserUtility
+import io.shiftleft.codepropertygraph.generated.{Cpg, EdgeTypes}
+import io.shiftleft.codepropertygraph.generated.nodes.{AstNodeNew, File, NewDbNode, NewTag}
+import io.shiftleft.semanticcpg.language.*
+import ai.privado.languageEngine.java.language.*
+
+import scala.collection.immutable.{HashMap, HashSet}
+import scala.collection.mutable
+
+class RubyMongoSchemaMapper(cpg: Cpg, ruleCache: RuleCache) extends PrivadoSimpleCpgPass(cpg) {
+
+  override def run(builder: DiffGraphBuilder): Unit = {
+    addToDatabaseCache(builder)
+  }
+  def addToDatabaseCache(builder: DiffGraphBuilder): Unit = {
+    /* Look for property nodes which match the regex `.*clients.*database` and are present in the file `config/mongoid.yaml`
+      These gives us all the client nodes defined for the project
+
+      production:
+        <<: *base_mongoid_specific_options
+        clients:
+          default:
+            database: default_production
+          noncustomer_default:
+            database: noncustomer_production
+
+      Here we are interested in property nodes with values `default_production` and `noncustomer_production`
+     */
+    val clientNodes =
+      cpg.property.where(_.file.name(".*config/mongoid[.](yml|yaml)")).where(_.name(".*clients.*database")).l
+
+    val deploymentTypeSet = mutable.HashSet[String]()
+    val clientsSet        = mutable.HashSet[String]()
+
+    clientNodes.foreach(client => {
```

Comment:
Instead of using a foreach here, can we not use a `ParalleCpgPass`?

Reply:
The reason I remember is, there is dependency of clientNodes to build `deploymentTypeSet`, `clientsSet` and few other operations, like clientNodes is again used on line number 94 to generate the db information and Database details cache

Reply:
There is possibility but that will require the pass to be divided into more granular passes and refactoring. Which we can focus on in near future